cluster/network.go

```diff
@@ -209,3 +231,236 @@ func (c *Cluster) getNetworkPluginManifest(pluginConfig map[string]string) (stri
 		return "", fmt.Errorf("[network] Unsupported network plugin: %s", c.Network.Plugin)
 	}
 }
+
+func (c *Cluster) CheckClusterPorts(ctx context.Context) error {
+	if err := c.deployTCPPortListeners(ctx); err != nil {
+		return err
+	}
+	if err := c.runServicePortChecks(ctx); err != nil {
+		return err
+	}
+	if err := c.checkKubeAPIPort(ctx); err != nil {
+		return err
+	}
+	return c.removeTCPPortListeners(ctx)
+}
+
+func (c *Cluster) checkKubeAPIPort(ctx context.Context) error {
+	log.Infof(ctx, "[network] Checking KubeAPI port Control Plane hosts")
+	for _, host := range c.ControlPlaneHosts {
+		logrus.Debugf("[network] Checking KubeAPI port [%s] on host: %s", KubeAPIPort, host.Address)
+		address := fmt.Sprintf("%s:%s", host.Address, KubeAPIPort)
+		conn, err := net.Dial("tcp", address)
+		if err != nil {
+			return fmt.Errorf("[network] Can't access KubeAPI port [%s] on Control Plane host: %s", KubeAPIPort, host.Address)
+		}
+		conn.Close()
```

Comment:
should we defer conn.Close()? in case error happens, is the conn gonna be nil? if not, we should defer closing it.

Reply:
If there is an error, conn is nil and the defer panics, so this should be fine.